src/services/extensions/links.ts

```diff
@@ -0,0 +1,109 @@
+import { PluginExtensionLinkConfig, PluginExtensionPanelContext, PluginExtensionPoints } from '@grafana/data';
+import { LabelType } from 'services/fields';
+import { getMatcherFromQuery } from 'services/logql';
+
+import { LokiQuery } from 'services/query';
+import { appendUrlParameter, createAppUrl, setUrlParameter, UrlParameterType } from 'services/routing';
+
+const title = 'Open in Explore Logs';
+const description = 'Open current query in the Explore Logs view';
+const icon = 'gf-logs';
+
+export const linkConfigs: PluginExtensionLinkConfig[] = [
+  {
+    extensionPointId: PluginExtensionPoints.DashboardPanelMenu,
+    title,
+    description,
+    icon,
+    path: createAppUrl(),
+    configure: (context?: PluginExtensionPanelContext) => {
+      if (!context) {
```

Comment:
Lines 20:40 and 66:86 seem to be doing the same thing. Maybe extract to a common place?

Reply:
Sounds good!